src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.CodeGeneration.targets

```diff
@@ -0,0 +1,126 @@
+<Project>
+  <!--
+    Targets used for Razor SDK code generation. Support for the RazorCoreGenerate target.
+  -->
+
+  <!-- 
+    Consider these properties to be private to this targets file. The main Razor SDK should define all of the properties
+    that we use to pass data back and forth. 
+  -->
+  <PropertyGroup>
+    <!-- Used for tag helper discovery -->
+    <_RazorTagHelperInputCache>$(IntermediateOutputPath)$(TargetName).TagHelpers.input.cache</_RazorTagHelperInputCache>
+    <_RazorTagHelperOutputCache>$(IntermediateOutputPath)$(TargetName).TagHelpers.output.cache</_RazorTagHelperOutputCache>
+    
+    <!-- Used to locate our tools -->
+    <_RazorGenerateToolAssembly>$(_RazorMSBuildRoot)tools\Microsoft.AspNetCore.Razor.GenerateTool.dll</_RazorGenerateToolAssembly>
+    <_RazorTagHelperToolAssembly>$(_RazorMSBuildRoot)tools\Microsoft.AspNetCore.Razor.TagHelperTool.dll</_RazorTagHelperToolAssembly>
+
+    <!-- Used to hash file inputs for RazorGenerate -->
+    <_RazorGenerateInputsHash></_RazorGenerateInputsHash>
+    <_RazorGenerateInputsHashFile>$(IntermediateOutputPath)$(MSBuildProjectName).RazorCoreGenerate.cache</_RazorGenerateInputsHashFile>
+  </PropertyGroup>
+
+  <!--
+    This target will only be called when we have some .cshtml files that are going to participate in code generation.
+
+    This is part of the chain of targets that are called once we've actually committed to generating code.
+  -->
+  <Target 
+    Name="_HashRazorGenerateInputs" 
+    AfterTargets="ResolveRazorGenerateInputs"
+    Condition="'@(RazorGenerate)'!=''">
+
+    <Hash ItemsToHash="@(RazorGenerate)" Condition="'@(RazorGenerate)'!=''">
```

Comment:
You can remove the `Condition` here since it's already being checked in the target.